keps/sig-apps/2232-suspend-jobs/README.md

```diff
@@ -385,78 +389,84 @@ field.
+  should be observed. Unexpected spikes in the metric with label
+  `action=pods_deleted` is potentially an indicator that:
+    1. Jobs are getting suspended when they shouldn't be.
+    2. Job sync latency is high when Job are suspended.
+  While the above list isn't exhaustive, they're signals in favour of rollbacks.
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path
+  tested?** <!-- I'll answer this after implementation.
   Describe manual testing that was done and the outcomes.
   Longer term, we may want to require automated upgrade/rollback tests, but we
-  are missing a bunch of machinery and tooling and can't do that now.
+  are missing a bunch of machinery and tooling and can't do that now. -->
 
-* **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
-fields of API types, flags, etc.?**
-  Even if applying deprecation policies, they may still surprise some users.
+* **Is the rollout accompanied by any deprecations and/or removals of features,
+  APIs, fields of API types, flags, etc.?** No.
 
 ### Monitoring Requirements
 
 _This section must be completed when targeting beta graduation to a release._
 
 * **How can an operator determine if the feature is in use by workloads?**
-  Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
-  checking if there are objects with field X set) may be a last resort. Avoid
-  logs or events for this purpose.
-
-* **What are the SLIs (Service Level Indicators) an operator can use to determine 
-the health of the service?**
-  - [ ] Metrics
-    - Metric name:
-    - [Optional] Aggregation method:
+  The `.spec.suspend` field is set to true by Jobs. The status conditions of a
+  Job can also be used to determine whether a Job is using the feature (look
+  for a condition of type "Suspended").
+
+* **What are the SLIs (Service Level Indicators) an operator can use to
+  determine the health of the service?**
+  - [x] Metrics
+    - Metric name: `job_sync_duration_seconds` gets a new label named `action`
+      to allow operators to filter Job sync latency by the action performed.
+      The possible values for this label are:
+        - `reconciling` when the Job's pod creation/deletion expectations are
+          unsatisfied and the controller is waiting for issued Pod
+          creation/deletions to complete.
+        - `tracking` when the Job's pod creation/deletion expectations are
+          satisfied and the number of active Pods matches expectations (i.e. no
+          pod creation/deletions issued in this sync). This is expected to be
+          the most common action in most of the syncs.
+        - `pods_created` - when the controller creates Pods. This can happen
+          when the number of active Pods is less than the wanted Job
+          parallelism.
+        - `pods_deleted` when the controller deletes Pods. This can happen if a
```

Comment:
in indexed job mode, isn't it possible to delete and create pods in the same sync cycle (in which case pods_created and pods_deleted wouldn't be mutually exclusive)?

Reply:
We agreed to only do delete or create, but not both. Currently, both operations are only possible if there are duplicated indexes. Since this would be such a rare circumstance, the change can be seen as noop.

Reply:
Correct. For posterity, we also decided against multiple label values in a single sample to keep things simple.